tests/projects/test_project.py

```diff
@@ -14,25 +15,41 @@
 
 def test_sync_functions():
     project_name = "project-name"
-    project = mlrun.new_project(project_name, skip_save=True)
+    project = mlrun.new_project(project_name, save=False)
     project.set_function("hub://describe", "describe")
     project_function_object = project.spec._function_objects
     project_file_path = pathlib.Path(tests.conftest.results) / "project.yaml"
     project.export(str(project_file_path))
-    imported_project = mlrun.load_project("./", str(project_file_path))
+    imported_project = mlrun.load_project("./", str(project_file_path), save=False)
     assert imported_project.spec._function_objects == {}
     imported_project.sync_functions()
     _assert_project_function_objects(imported_project, project_function_object)
 
-    fn = project.func("describe")
+    fn = project.get_function("describe")
     assert fn.metadata.name == "describe", "func did not return"
 
     # test that functions can be fetched from the DB (w/o set_function)
     mlrun.import_function("hub://sklearn_classifier", new_name="train").save()
-    fn = project.func("train")
+    fn = project.get_function("train")
     assert fn.metadata.name == "train", "train func did not return"
 
 
+def test_load_save_project(monkeypatch):
+    project_name = "project-name"
+    project = mlrun.new_project(project_name, save=False)
+    project.set_function("hub://describe", "describe")
+    project_file_path = pathlib.Path(tests.conftest.results) / "project.yaml"
+
+    project.save = unittest.mock.Mock()
+    monkeypatch.setattr(
+        mlrun.projects.project, "_load_project_file", lambda *args, **kwargs: project
+    )
+
+    loaded_project = mlrun.load_project("./", str(project_file_path), save=True)
+    assert project.save.call_count == 1
```

Comment:
This is testing code instead of testing behavior
You don't want to test that `load_project` calls `save` you simply want to test that `load_project` works, it didn't when it was registering artifacts, test that.
Should probably an integration test in order to test that properly